### PR TITLE
draft cli post

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

**Comment by maelle** [comment]
Any further comment @gaborcsardi? :smile_cat: (no pressure though)

File: content/post/2023-10-20-cliff-notes-about-cli/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+cli::cli_alert_success("marvellous package adopted!")
+```
+
+In this post, we transform the hurdles we encountered in a series of tips so that your own journey might be slightly smoother, and also to encourage you to try out cli in your package!
+
+## cli is the thing for package interfaces now!
+
+You can view cli as a domain-specific language (DSL) for command-line interfaces (CLI): Just like tidyverse makes your data pipelines easier to construct and more readable, cli makes your communication producing code simpler to write!
+
+cli, well its author Gábor Csárdi, deals with pesky details and we package developers only need to use the high-level interface.
```
Comment (gaborcsardi) [inline content/post/2023-10-20-cliff-notes-about-cli/index.Rmd:66]:
```suggestion
cli deals with pesky details and we package developers only need to use the high-level interface.
```

Appreciate the shoutout, but it is team work. :)

File: content/post/2023-10-20-cliff-notes-about-cli/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+Michael McCarthy [recommends](https://mastodon.social/@mccarthymg@fosstodon.org/110673498809652527) looking at what cli itself does.
+
+### What about dark themes?
+
+Together, we cover both sides of having dark (Mo) and light (Maëlle) themes in RStudio.
+It would be easy to assume that this all would not work well on dark theme, but Mo can tell you that it all works seamlessly.
+The output colors change to be dark friendly!
+For that reason, user-overrides of colors can be a tricky thing, as you are no longer relying on the excellent tooling of cli to make sure this works in both light and dark mode.
+
+Open questions remain: How to let an user override it with their own?
```
Comment (gaborcsardi) [inline content/post/2023-10-20-cliff-notes-about-cli/index.Rmd:139]: You can define your own themes, e.g. put this in your profile to make your variables underlined:
```r
options(cli.user_theme = list(.var = list("text-decoration" = "underline")))
cli::start_app() # not needed if the previous line is in the profile
cli::cli_text("Don't call a variable {.var variable}!")
```

**Review by gaborcsardi — APPROVED** [review]
Great set of tips, thank you for doing this! Left some minor suggestions.

**Comment by maelle** [comment]
Thanks a ton for reviewing @gaborcsardi!

File: content/post/2023-11-27-cliff-notes-about-cli/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+It would be easy to assume that this all would not work well on dark theme, but Mo can tell you that it all works seamlessly.
+The output colors change to be dark friendly!
+For that reason, user-overrides of colors can be a tricky thing, as you are no longer relying on the excellent tooling of cli to make sure this works in both light and dark mode.
+
+An user can define their own themes, for instance putting this in their profile to make variables underlined:
+
+```{r, eval=FALSE}
+options(cli.user_theme = list(.var = list("text-decoration" = "underline")))
+cli::start_app() # not needed if the previous line is in the profile
+cli::cli_text("Don't call a variable {.var variable}!")
```
[the remark this answers is not in the document]
Comment (gaborcsardi) [reply content/post/2023-11-27-cliff-notes-about-cli/index.Rmd:144]: Btw. you can also change what cli thinks about the base ANSI colors, i.e. what is "red", by choosing a palette: 
https://cli.r-lib.org/reference/ansi_palettes.html?q=palette
https://cli.r-lib.org/articles/palettes.html

[the remark this answers is not in the document]
Comment (drmowinckels) [reply content/post/2023-11-27-cliff-notes-about-cli/index.Rmd:144]: thanks @gaborcsardi ! That put things into the necessary context. I'm gonna suggest a little edit to make that more obvious here.

**Pull request merged**